src/tree/hist/histogram.h

```diff
@@ -180,47 +221,100 @@ class HistogramBuilder {
 
     common::BlockedSpace2d space(
         nodes_to_build.size(), [&](std::size_t) { return n_total_bins; }, 1024);
-    common::ParallelFor2d(space, this->n_threads_, [&](size_t node, common::Range1d r) {
+    common::ParallelFor2d(space, this->n_threads_, [&](std::size_t node, common::Range1d r) {
       // Merging histograms from each thread.
       this->buffer_.ReduceHist(node, r.begin(), r.end());
```

Comment:
Is this reduce hist function call actually useful for secure training?

Reply:
I think so, this is the multi-thread part for each client

Reply:
I thought the histogram is being built in the plugin? If so, any operation for the histogram in XGBoost should be invalid right?

Reply:
ah you are right! 
this reduce hist will have no impact for secure vertical training, it will be needed by secure horizontal, and non-secure federated schemes.